simphony/io/data_container_table.py

```diff
@@ -0,0 +1,169 @@
+
+        """
+        handle = root._v_file
+        self._parent = parent = root
+
+        if hasattr(parent, name):
+            self._table = getattr(parent, name)
+        else:
+            self._table = handle.create_table(parent, name, Record)
+
+        # prepare useful mappings
+        columns = Record.columns['Data']._v_colobjects
+        members = CUBA.__members__
+        self._cuba_to_position = {
+            cuba: columns[member.lower()]._v_pos
+            for member, cuba in members.items()}
+        self._cuba_to_column = {
+            cuba: member.lower()
+            for member, cuba in members.items()}
+        self._position_to_cuba = {
+            columns[member.lower()]._v_pos: cuba
+            for member, cuba in members.items()}
+
+    def append(self, data):
+        """ Append the data to the end of the table.
+
+        Parameters
+        ----------
+        data : DataContainer
+            The DataContainer instance to save.
+
+        Returns
+        -------
+        uid : uuid.UUID
+            The index of the saved row.
+
+        """
+        table = self._table
+        positions = self._cuba_to_position
+        row = table.row
+        mask = numpy.zeros(
+            shape=table.coldtypes['mask'].shape, dtype=numpy.bool)
+        uid = uuid.uuid4()
+        row['index'] = uid.bytes
+        data_row = list(row['Data'])
+        for key in data:
+            data_row[positions[key]] = data[key]
+            mask[positions[key]] = True
+        row['mask'] = mask
+        row['Data'] = tuple(data_row)
+        row.append()
+        table.flush()
+        return uid
+
+    def __getitem__(self, uid):
+        """ Return the DataContainer in row.
+
+        """
+        cuba = self._position_to_cuba
+        for row in self._table.where(
+                'index == value',  condvars={'value': uid.bytes}):
+            mask = row['mask']
+            data = row['Data']
+            return DataContainer({
+                cuba[index]: data[index]
+                for index, valid in enumerate(mask) if valid})
+
+    def __setitem__(self, uid, data):
+        """ Set the data in row from the DataContainer.
+
+        """
+        table = self._table
+        positions = self._cuba_to_position
+        for row in table.where(
+                'index == value', condvars={'value': uid.bytes}):
+            mask = numpy.zeros(
+                shape=table.coldtypes['mask'].shape, dtype=numpy.bool)
+            data_row = list(row['Data'])
+            for key in data:
+                data_row[positions[key]] = data[key]
+                mask[positions[key]] = True
+            row['mask'] = mask
+            row['Data'] = tuple(data_row)
+            row.update()
+            # see https://github.com/PyTables/PyTables/issues/11
+            row._flush_mod_rows()
+            break
+        else:
+            raise KeyError('Index {} is not found'.format(uid))
```

Comment:
I think that raising an error here is incorrect since the code
```python
uid = uuid.uuid4()
data_container_table[uid] = DataContainer()
```
should add an empty row in the table mapped to uid instead of raising and error. This will make using the DataContainerTable inside the CUDS container much more convenient.